### PR TITLE
Random heal target new ability

### PR DESCRIPTION
**Pull request by MatthewEpperson**

### CODE REVIEW

File: Gaming The Syllabus/Assets/XML/Abilities/PlayerAbilities.xml [hunk 1]
```diff
@@ -2,28 +2,30 @@
...
-        <effect type="DamageTarget" target="AnyEnemy" damageAmount="5"/>
+        <effect type="RandomDamageTarget" target="AnyEnemy" minDamage="8" maxDamage="16"/>
     </Effects>
     <hitMessage> Used Fire Blast and dealt </hitMessage>
     <missMessage> Tried using Fire Blast but missed! </missMessage>
   </PlayerAbility>
-  <!-- <PlayerAbility name="Super Attack" type="attack" default="false">
-    <AbilityDesc> This is a super attack that deals high damage but at the cost of some of your HP.</AbilityDesc>
-  </PlayerAbility> -->
+  <PlayerAbility name="Damaging Regeneration" type="attack" default="true">
```
Comment (travis-bittel) [inline Gaming The Syllabus/Assets/XML/Abilities/PlayerAbilities.xml:21]: This looks awesome! Glad we got everything using the ActionEffects system and excited to see what other abilities we can come up with!

File: Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            int healAmount = calculateHealAmount(minHeal, maxHeal);
+            target.HealUnit(healAmount);
+            DialogueBoxUIManager.AddStringToDialogueBox
+            ($"{DialogueBoxUIManager.FormatCombatUnitColor(caller)}"
+            + $"{DialogueBoxUIManager.FormatDamageColor(healAmount)}");
+        }
+    }
+
+    private int calculateHealAmount(int minHeal, int maxHeal)
```
Comment (travis-bittel) [inline Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs:39]: PascalCase for method names

File: Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+using System.Runtime.Versioning;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Xml.Serialization;
+using System.Xml.Schema;
+using System.Xml;
+
+public class RandomHealTarget : ActionEffect
```
Comment (travis-bittel) [inline Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs:9]: Looks awesome!

File: Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    public int maxHeal { get; private set; }
+
+    public RandomHealTarget(int minHeal, int maxHeal, TargetType target)
+    {
+        this.minHeal = minHeal;
+        this.maxHeal = maxHeal;
+        Target = target;
+    }
+
+    public RandomHealTarget()
```
Comment (travis-bittel) [inline Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs:21]: Annoys me to no end that we have to have this no-args constructor around for XML Serialization... wonder if there is a way around that

**Review by travis-bittel — APPROVED** [review]

File: Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs (hunk 3 above)
Comment (rjmm99) [reply Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs:9]: +1 Looks great!

File: Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        System.Random randomNum = new System.Random();
+        return randomNum.Next(minHeal, maxHeal + 1);
+    }
+
+    public override void WriteXml(XmlWriter writer)
+    {
+        base.WriteXml(writer);
+        writer.WriteAttributeString("minHeal", minHeal.ToString());
+        writer.WriteAttributeString("maxHeal", maxHeal.ToString());
+    }
```
Comment (Isildur512) [inline Gaming The Syllabus/Assets/Scripts/Combat/Action Effects/RandomHealTarget.cs:50]: I like how well implemented the XML fuctionality is here. It was a good way to approach this system.

**Pull request merged**